lib/ansible/modules/cloud/amazon/cloudfront_distribution.py

```diff
@@ -0,0 +1,2106 @@
+
+DOCUMENTATION = '''
+---
+
+module: cloudfront_distribution
+
+short_description: create, update and delete aws cloudfront distributions.
+
+description:
+    - Allows for easy creation, updating and deletion of CloudFront
+      distributions.
+
+requirements:
+  - boto3 >= 1.0.0
+  - python >= 2.6
+
+version_added: "2.4"
+
+author: Willem van Ketwich (@wilvk)
+
+options:
+
+    state:
+      description:
+        - The desired state of the distribution or streaming distribution.
+          present - creates a new distribution or updates an existing
+            distribution.
+          absent - deletes an existing distribution.
+      choices: ['present', 'absent']
+      required: false
+
+    distribution_id:
+      description:
+        - The id of the cloudfront distribution.
+          This parameter can be exchanged with C(alias) or C(caller_reference)
+          and is used in conjunction with C(e_tag).
+          Used with C(streaming_distribution=no)
+      required: false
+
+    streaming_distribution_id:
+      description:
+        - The id of the CloudFront streaming distribution.
+          This parameter can be exchanged with C(alias) and is used in
+          conjunction with C(e_tag).
+          Used with C(streaming_distribution=yes).
+      required: false
+
+    e_tag:
+      description:
+        - A unique identifier of a modified or existing distribution.
+          Used in conjunction with C(distribution_id) or
+          C(streaming_distribution_id).
+          Is determined automatically if not specified.
+      required: false
+
+    generate_presigned_url:
+      description:
+        - Generates a presigned url for a distribution from a private
+          I(.pem) file key.
+      default: 'no'
+      choices: ['yes', 'no']
+      required: false
+
+    caller_reference:
+      description:
+        - A unique identifier for creating and updating cloudfront
+          distributions. Each caller reference must be unique across all
+          distributions. eg. a caller reference used in a web distribution
+          cannot be reused in a streaming distribution.
+          This parameter can be used instead of C(distribution_id) or
+          C(streaming_distribution_id) to reference an existing distribution.
+          If not specified, this defaults to a datetime stamp of the format
+            'YYYY-MM-DDTHH:MM:SS.ffffff'
+      required: false
+
+    pem_key_path:
+      description:
+        - The path on the host to the I(.pem) private key file.
+          This key is used to sign the url.
+      required: false
+
+    pem_key_password:
+      description:
+        - The optional password for the pem private key file.
+      required: false
+
+    cloudfront_url_to_sign:
+      description:
+        - The cloudfront url to sign with the pem private key.
+      required: false
+
+    presigned_url_expire_date:
+      description:
+        - The expiry date of the presigned url. Date format is I(YYYY-MM-DD)
```

Comment:
YYYY-MM-DD isn't a module option, so I think you can remove the I and parentheses